### PR TITLE
Adds lung damage, reworks smoking

### PR DESCRIPTION
**Pull request by RandolfTheMeh**
## About The Pull Request

I wanted to add lung damage once I found out there was no such mechanic yet. Given the recent attention given to medical in regard to changes, I felt it necessary to give lungs- a critical part of the body's functions- a means of being damaged. The one difference, however, is that they don't have a binary work/don't work state. Instead, as they get damaged, they require more partial O2. With standard lungs, a healthy set can breath 16 kPa just fine- but a damaged set will go upwards to 21 kPa, before failing completely.

I then added the nicotine addiction's new effects, generating small amounts of lung damage. It was difficult getting numbers for this, and I still feel it requires tweaking. Nicotine, unlike the other narcotics we have, has no withdrawal symptoms. Now you'll experience small amounts of toxic buildup, and lung damage. The two are rather inconsequential, and so long as you keep feeding your addiction, will prevent the toxic buildup from occurring, and the lung damage from getting too severe (though as it is now, I feel it is rather tame, so the numbers will definitely need to be tweaked). As a result of this, the Smoker trait now gives 2 points instead of 1, since there is a physical detriment to your character now in taking it.

Edit: Nicotine addiction no longer causes the lung damage, but instead smoking does. The amount is minor, and would require something like 45 minutes of chain smoking to end up with a failing lung. Lungs heal similarly to livers, too, and as they heal so too does the safe amount of partial O2 decrease.

## Why It's Good For The Game

I feel that with the current interest in medical reworks, it is necessary we have an organ damage mechanic for all our major organs. Though this PR will likely not impact an individual's round whatsoever except in some fringe circumstances, I feel it lays the groundwork that'll allow other PRs to do interesting things with anything pertaining to medical, such as chemicals that result in organ damage/failure as opposed to just dealing toxin damage, or surgical mistakes resulting in a damaged organ if the operation deals with a location that has any, etc..

## Changelog
:cl:
add: Lung damage and lung failure. As lung damage increases, so too does the partial pressure of O2 you need in order to breath properly. For organic lungs, this range is from 16 kPa to 21 kPa. Lung failure will prevent you from breathing altogether.
add: Organ reconstruction. An operation that repairs damaged portions of the liver and lungs, so long as the patient has synthflesh within their bloodstream. Careful, this operation will not work on organs that are already failing, and without synthflesh within their bloodstream, attempting to repair these organs will only damage it further!
balance: Smoking something causes small amounts of lung damage, and to compensate the Smoker trait has been boosted from 1 to 2 points.
/:cl:

### CODE REVIEW

**Comment by Swiftkor** [comment]
sounds good, you may want to reduce the threshold for nicotine addiction so that you can get addicted without specifically trying to get addicted.

**Comment by RandolfTheMeh** [comment]
A good point. I'll have to do some more edits tomorrow, and I'll be sure to look into that. I'm still uncertain with just how much lung damage a nicotine addiction should result in.

**Comment by HomocidlePancakes** [comment]
You don’t always get nicotine via smoking. Doesn’t make sense really that you get lung damage from a nicotine patch/chewing tobacco

**Comment by Krysonism** [comment]
Organ damage is something i have liked to see for a long time.

I also would like to see organ damage and traums added for all the other organs in the future.

**Comment by Shapsy** [comment]
I like how the best way to avoid your lungs shutting down is to continue smoking as much as possibke

**Comment by CRITAWAKETS** [comment]
You should make it where breathing hot gas or breathing a large enough amount of water vapor (and maybe plasma?) causes lung damage, as would it in real-life.

**Comment by py01** [comment]
Have you added a way to repair lung damage?

**Comment by HomocidlePancakes** [comment]
We don't have a way to repair liver damage too and having a drug that would heal lungs would 100% nullify this feature

**Comment by Tlaltecuhtli** [comment]
s u r g e r y

**Comment by HomocidlePancakes** [comment]
Yes Tlal, that exists. But clearly the lad meant a specific chemical to heal lungs, and because  of that I mentioned only chems.

**Comment by ATH1909** [comment]
We should add surgeries that can heal organs that haven't failed/that aren't too damaged, yeah.

According to my calculations, trying to kick a nicotine addiction should deal only 37 points of lung damage (if you don't take any nicotine while trying to kick the addiction). Is that intentional? IMO, you should get minor amounts of lung damage for having nicotine in your system (or from something to do with reaction_mob?), no?

**Comment by RandolfTheMeh** [comment]
The low damage is intentional, yes. Doing 37 lung damage means your breathable partial pressure of O2 becomes 17-18. I've debated tying lung damage to nicotine, but I felt it more fairly tied to its addiction stages, since any amount of lung damage means you can no longer breath 16 kPa, and thus have to modify your internals release pressure. Having that tied to any consumption of nicotine I feel will make it more common than I want it to be, since this is supposed to be a minor detriment that can be exacerbated by other lung-damaging sources to cause lung failure. The goal for balance was for lung failure to occur within the 1hr - 1.5hr mark if an individual takes the smoker trait.

**Comment by RandolfTheMeh** [comment]
I'm still open to the idea of nicotine causing lung damage, but if I do so, I may change the default release pressure of tanks back to 21 kPa so newer players don't suffocate on their own internals tanks.

**Comment by anconfuzedrock** [comment]
>21 kpa

are we going to go back to 21 being the base oxytank release

**Comment by Mickyan** [comment]
Lung damage on addiction is kind of bizarre, especially since addiction stages are withdrawal so.. you have to smoke more to not get lung damage?

**Comment by Tlaltecuhtli** [comment]
i think its better to leave it as only replaceable, gives more incentives into researching cyber  organs/ harvest organs

**Comment by Swiftkor** [comment]
Yeah I definitely think organ transplants should be needed more and organ damage is a good way of doing that

**Comment by nero1024** [comment]
Isn't smoking itself supposed to cause lung damage? Nicotine is only supposed to be addictive, why should it cause lung damage?

**Comment by Krysonism** [comment]
It seems like it would be better to let the cigarette itself do the lung damage under the process() proc

It would be easy give different lung damage values for cigarette subtypes.

If you go this route Shady Jim's should probably do more damage and cigars do less.

**Comment by RandolfTheMeh** [comment]
I dislike the idea of lung damage being tied to the act of smoking/nicotine from a balance perspective. I'm fine with changing it to smoking more than to nicotine, but I was hoping to make it act as a parallel to liver damage and alcohol, in that over-consumption would be what triggers the organ damage, allowing for small bouts of use, so that the mechanic of drinking/smoking is not altogether useless.
So the change I'll work on is for smoking to cause lung damage, in very, very minor amounts. Something around 45 minutes would cause total lung failure if chain-smoking.
The reason it's so minor is that I want to add organ damage to the heart and other organs that don't have it yet (though to my knowledge, most already do, so I'll have to check what other important organs are needed. Stomach maybe?) , so that when dead your organs suffer damage. When you've been dead for ten minutes, your heart's been damaged to the point of no return, or your brain's been damaged to the point of no return, you'll no longer be able to be resuscitated. Thus by smoking, you increase the likelihood you'll need a lung transplant upon death, and by smoking nicotine specifically, you lower the time in which you can be effectively defibbed.

tl;dr since most of this isn't relevant, I'll be adding the lung damage proc on smoke, increasing internals release pressure to 21 kPa, and lowering the smoker trait back down to 1 point until I can implement heart damage, if such a change is accepted.

**Comment by Mickyan** [comment]
For the liver any amount of alcohol damages it (exponentially scaling with volume and power) but the damage slowly heals over time, you may want to do the same here if that's what you're going for

**Comment by kittymaster0** [comment]
So if i inject nicotine my lungs die? o...kay.

**Comment by RandolfTheMeh** [comment]
Changes have been made. Safe O2 minimum doesn't go back to its original value due to a rounding error- if the check came prior to lowering damage to 0, we'd go below our previous safe O2 minimum, which is less preferable than being .002 points above the original point. Standard release pressure is now 21 kPa so the newfriends don't suffocate themselves after taking a cigarette.

**Comment by Tlaltecuhtli** [comment]
>Standard release pressure is now 21 kPa so the newfriends don't suffocate themselves after taking a cigarette.

:/

**Comment by Akrilla** [comment]
Buff cigs

**Comment by Shadowflame909** [comment]
Can Robotic Lungs repair lung damage?

Robotic organs are supposed to be the medicinal end-game I believe.

**Comment by RandolfTheMeh** [comment]
Cybernetic lungs work similarly to cybernetic livers. They heal at the same rate as their organic counterparts, but have more health, and take damage on being EMPed.

**Comment by Razharas** [comment]
Slowly but surely, organ based health-damage system is becoming a reality
Just like planned for many years already
Beautiful pr, a seal of approval from me

**Comment by Michael-Ashfield** [comment]
How will this affect Space tobacco, which is supposed to heal your breathing. Will it still damage the lungs or might it get a botany only lung healing chemical (departmental co-operation).

**Comment by optimumtact** [comment]
it's fine to have no chemical healing, surgery will suffice, along with replacement organs

**Comment by ExcessiveUseOfCobblestone** [comment]
ugh I really want a proper organ damage system before we start taping it on piecewise for each organ.

**Comment by ExcessiveUseOfCobblestone** [comment]
> How will this affect Space tobacco, which is supposed to heal your breathing. Will it still damage the lungs or might it get a botany only lung healing chemical (departmental co-operation).

It still does, but it won't touch the long-term effect so you'll need to dope up on it if you take enough damage.

The solution is to not be in a situation where you're constantly taking lung damage lol.

**Comment by interestingusernam3** [comment]
Breathing in toxin smoke should also deal lung damage.

**Comment by RandolfTheMeh** [comment]
> 
> 
> ugh I really want a proper organ damage system before we start taping it on piecewise for each organ.

If you'd rather that, I can work on it. Lung damage was a thing I wrote for a feature that I can't see being merged, septicemia. I decided to PR just the lung damage with the intention of adding a more uniform organ damage system if the interest was there. If you'd rather this PR closed in favor of a full organ damage system, that's fine.

> 
>
> it's fine to have no chemical healing, surgery will suffice, along with replacement organs

Added a baseline surgery to repair liver/lung damage.

**Comment by Tlaltecuhtli** [comment]
check this idea for organ damage: give all organs a damage var and have them process() when damaged
the organ rolls for failure effects which the higher the damage the wider the effects can be like roll cap is damage/10 so when you have 20 damage you switch case prob between 1 and 2 which could be just flavor text which warn you about organ damage like "your heart hurts" while around 50 damage you roll between 1 and 5 and you can have 4 - 5 give more serious effects like stamina hits for heart, vomiting for stomach, then when you reach 100 damage you can roll 10 which is a super bad lethal effect like cardiac arrest or liver failure. chemicals like corazone can just lower the roll value of 2 so you can be sure to not get the lethal, but you still get affected by mild/strong debuffs and you ll need surgery to be fixed 100%
organs get damaged by bad behavior like taking stims continually starts damaging the heart, eating fried crowbars damages the stomach etc

**Comment by ExcessiveUseOfCobblestone** [comment]
It doesn't even need to *do* anything for the moment, just having a base to build off of instead of building a base off each individual organ would be extremely helpful you/anyone else interested in fleshing the feature out.

**Comment by RandolfTheMeh** [comment]
> 
> 
> It doesn't even need to _do_ anything for the moment, just having a base to build off of instead of building a base off each individual organ would be extremely helpful you/anyone else interested in fleshing the feature out.

In that case, I'll close this PR and work on a universal organ damage system.
